tests/test_measure.py

```diff
@@ -26,41 +47,54 @@ def get_meas_results(meas_function, cpus=1, measure_kwargs=None):
         shifts=shifts,
         indexes=indexes,
         stamp_size=stamp_size,
-        batch_size=1,
         seed=TEST_SEED,
     )
     meas_generator = MeasureGenerator(
         meas_function, draw_blend_generator, cpus=cpus, measure_kwargs=measure_kwargs
     )
     blend_results, results = next(meas_generator)
-    wcs = blend_results["wcs"]
-    x, y = wcs.world_to_pixel_values(shifts[0][0] / 3600, shifts[0][1] / 3600)
-    target = np.array([x.item(), y.item()])
+    target = np.array(
+        [[blend["x_peak"].item(), blend["y_peak"].item()] for blend in blend_results["blend_list"]]
+    )
     return target, results
 
 
 def compare_sep():
-    """Test detection with sep"""
-    target, results = get_meas_results(sep_measure, measure_kwargs=[{"sigma_noise": 2.0}])
-    x_peak, y_peak = (
-        results["catalog"]["sep_measure"][0]["x_peak"].item(),
-        results["catalog"]["sep_measure"][0]["y_peak"].item(),
+    """Test sep detection using single band and multiband"""
+    target, results = get_meas_results(
+        [sep_singleband_measure, sep_multiband_measure], measure_kwargs=[{"sigma_noise": 1.5}]
     )
-    detected_centers = np.array([x_peak, y_peak])
-    dist = np.max(np.abs(detected_centers - target))
-    np.testing.assert_array_less(dist, 1.0)
+    detected_sources = {"sep_singleband_measure": 0, "sep_multiband_measure": 0}
+    for meas_function in detected_sources.keys():
+        for i, blend in enumerate(results["catalog"][meas_function]):
+            if len(blend) > 0:
+                detected_centers = np.array([blend[0]["x_peak"].item(), blend[0]["y_peak"].item()])
+                dist = np.max(np.abs(detected_centers - target[i]))
+                np.testing.assert_array_less(dist, 1.5)
+                detected_sources[meas_function] += 1
+
+    assert detected_sources["sep_multiband_measure"] >= 0.5 * len(target)
+    assert detected_sources["sep_singleband_measure"] >= 0.5 * len(target)
+    assert detected_sources["sep_multiband_measure"] >= detected_sources["sep_singleband_measure"]
 
 
 def compare_sep_multiprocessing():
-    """Test detection with sep"""
-    target, results = get_meas_results(sep_measure, measure_kwargs=[{"sigma_noise": 2.0}], cpus=4)
-    x_peak, y_peak = (
-        results["catalog"]["sep_measure"][0]["x_peak"].item(),
-        results["catalog"]["sep_measure"][0]["y_peak"].item(),
+    """Test sep dettection using single band and multiband with multiprocessing"""
+    target, results = get_meas_results(
+        [sep_singleband_measure, sep_multiband_measure], measure_kwargs=[{"sigma_noise": 1.5}]
     )
-    detected_centers = np.array([x_peak, y_peak])
-    dist = np.max(np.abs(detected_centers - target))
-    np.testing.assert_array_less(dist, 1.0)
+    detected_sources = {"sep_singleband_measure": 0, "sep_multiband_measure": 0}
+    for meas_function in detected_sources.keys():
```

Comment:
could you add a small comment explaning what this block that follows is doing exactly?

Reply:
I was verifying that `sep_multiband_measure` detected more sources than `sep_singleband_measure`. I also was trying to check that each prediction was close to the truth. Let me know if there is a better way of testing, or if you have any specific tests in mind.

Reply:
now I get it, thanks! LGTM